extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+	 */
+	function loadFromLocal( $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$filename = $this->mLocalMessageCache . '/' . substr( $wgDefaultMessagesDB, 0, 1 ) . '/' . $wgDefaultMessagesDB . "/messages-$code";
+
+		# Check file existence
+		wfSuppressWarnings();
+		$file = fopen( $filename, 'r' );
+		wfRestoreWarnings();
+		if ( !$file ) {
+			return false; // No cache file
+		}
+
+		if ( $this->mLocalMessageCacheSerialized ) {
+			// Check to see if the file has the hash specified
+			$localHash = fread( $file, 32 );
+			if ( $hash === $localHash ) {
+				// All good, get the rest of it
+				$serialized = '';
+				while ( !feof( $file ) ) {
+					$serialized .= fread( $file, 100000 );
+				}
+				fclose( $file );
+				return $this->setCache( unserialize( $serialized ), $code );
+			} else {
+				fclose( $file );
+				return false; // Wrong hash
+			}
+		} else {
+			$localHash=substr(fread($file,40),8);
+			fclose($file);
+			if ($hash!=$localHash) {
+				return false; // Wrong hash
+			}
+
+			# Require overwrites the member variable or just shadows it?
+			require( $filename );
+			return $this->setCache( $this->mCache, $code );
+		}
+	}
+
+	/**
+	 * Save the cache to a local file.
+	 */
+	function saveToLocal( $serialized, $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$dirname = $this->mLocalMessageCache . '/' . substr( $wgDefaultMessagesDB, 0, 1 ) . '/' . $wgDefaultMessagesDB;
+		$filename = "$dirname/messages-$code";
+		wfMkdirParents( $dirname, 0777 ); // might fail
+
+		wfSuppressWarnings();
+		$file = fopen( $filename, 'w' );
+		wfRestoreWarnings();
+
+		if ( !$file ) {
+			wfDebug( "Unable to open local cache file for writing\n" );
+			return;
+		}
+
+		fwrite( $file, $hash . $serialized );
+		fclose( $file );
+		@chmod( $filename, 0666 );
+	}
+
+	function saveToScript( $array, $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$filename = $this->mLocalMessageCache . "/messages-$wgDefaultMessagesDB-$code";
+		$tempFilename = $filename . '.tmp';
+		wfMkdirParents( $this->mLocalMessageCache, 0777 ); // might fail
+
+		wfSuppressWarnings();
+		$file = fopen( $tempFilename, 'w');
+		wfRestoreWarnings();
+
+		if ( !$file ) {
+			wfDebug( "Unable to open local cache file for writing\n" );
+			return;
+		}
+
+		fwrite($file,"<?php\n//$hash\n\n \$this->mCache = array(");
+
+		foreach ($array as $key => $message) {
+			$key = $this->escapeForScript($key);
+			$messages = $this->escapeForScript($message);
+			fwrite($file, "'$key' => '$message',\n");
+		}
+
+		fwrite($file,");\n?>");
+		fclose($file);
+		rename($tempFilename, $filename);
+	}
+
+	function escapeForScript($string) {
+		$string = str_replace( '\\', '\\\\', $string );
+		$string = str_replace( '\'', '\\\'', $string );
+		return $string;
+	}
+
+	/**
+	 * Set the cache to $cache, if it is valid. Otherwise set the cache to false.
+	 */
+	function setCache( $cache, $code ) {
+		if ( isset( $cache['VERSION'] ) && $cache['VERSION'] == MSG_CACHE_VERSION ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "VERSION" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)